parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/Tokens.scala

```diff
@@ -35,4544 +73,47 @@ object Tokens extends CrfTokenable {
+  lazy val directions: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.direction"))
 
-  val flat: Seq[TokenIndicator] = Seq(
-    "FLAT", "FLT",
-    "APARTMENT", "APPTS", "APPT", "APTS", "APT",
-    "ROOM",
-    "ANNEX", "ANNEXE",
-    "UNIT",
-    "BLOCK", "BLK"
-  )
+  lazy val flat: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.flat"))
 
-  val company: Seq[TokenIndicator] = Seq(
-    "CIC",
-    "CIO",
-    "LLP",
-    "LP",
-    "LTD",
-    "LIMITED",
-    "CYF",
-    "PLC",
-    "CCC",
-    "UNLTD",
-    "ULTD"
-  )
+  lazy val company: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.company"))
 
-  val road: Seq[TokenIndicator] = Seq(
-    "ROAD", "RAOD", "RD",
-    "DRIVE", "DR",
-    "STREET", "STRT",
-    "AVENUE", "AVENEU",
-    "SQUARE",
-    "LANE", "LNE", "LN",
-    "COURT", "CRT", "CT",
-    "PARK", "PK",
-    "GRDN", "GARDEN",
-    "CRESCENT",
-    "CLOSE", "CL",
-    "WALK",
-    "WAY",
-    "TERRACE",
-    "BVLD",
-    "HEOL",
-    "FFORDD",
-    "PLACE",
-    "GARDENS",
-    "GROVE",
-    "VIEW",
-    "HILL",
-    "GREEN"
-  )
-  val residential: Seq[TokenIndicator] = Seq(
-    "HOUSE", "HSE",
-    "FARM",
-    "LODGE",
-    "COTTAGE", "COTTAGES",
-    "VILLA", "VILLAS",
-    "MAISONETTE",
-    "MEWS"
-  )
+  lazy val road: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.road"))
 
-  val business: Seq[TokenIndicator] = Seq(
-    "OFFICE",
-    "HOSPITAL",
-    "CARE",
-    "CLUB",
-    "BANK",
-    "BAR",
-    "UK",
-    "SOCIETY",
-    "PRISON",
-    "HMP",
-    "RC",
-    "UWE",
-    "UEA",
-    "LSE",
-    "KCL",
-    "UCL",
-    "UNI", "UNIV", "UNIVERSITY", "UNIVERISTY"
-  )
+  lazy val residential: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.residential"))
 
-  val locational: Seq[TokenIndicator] = Seq(
-    "BASEMENT", "GROUND", "ATTIC",
-    "UPPER", "ABOVE", "TOP", "LOWER", "FLOOR", "HIGHER",
-    "LEFT", "RIGHT", "FRONT", "BACK", "REAR",
-    "WHOLE", "PART", "SIDE"
-  )
+  lazy val business: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.business"))
+
+  lazy val locational: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.locational"))
+
+  lazy val ordinal: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.ordinal"))
+
+  lazy val outcodes: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.outcode"))
+
+  lazy val postTown: Seq[String] = fileToList(config.getString("parser.input-pre-post-processing.posttown"))
+
+  /**
+    * Contains key-value map of synonyms (replace key by value)
+    */
+  lazy val synonym: Map[String, String] = fileToList(config.getString("parser.input-pre-post-processing.synonym"))
+    // The "-1" in split is to catch the trailing empty space on the lines like "ENGLAND,"
+    .map(_.split(",", -1))
```

Comment:
Yes and also the desired behavior is that `"ENGLAND,"` returns `Array("ENGLAND", "")`